convert2rhel/actions/conversion/set_efi_config.py

```diff
@@ -0,0 +1,237 @@
+            ]
+            # without the required files user should not reboot the system
+            self.set_result(
+                level="ERROR",
+                id="UNABLE_TO_FIND_REQUIRED_FILE_FOR_GRUB_CONFIG",
+                title="Couldn't find system GRUB config",
+                description="Couldn't find one of the GRUB config files in the current system which is required for configuring UEFI for RHEL: {}".format(
+                    ", ".join(missing_files)
+                ),
+            )
+            return
+
+        for src_file in src_files:
+            # Skip non-existing file in destination directory
+            if not os.path.exists(src_file):
+                logger.debug(
+                    "The %s file does not exist in %s folder. Copying skipped."
+                    % (os.path.basename(src_file), CENTOS_EFIDIR_CANONICAL_PATH)
+                )
+                continue
+            # Skip already existing file in destination directory
+            dst_file = os.path.join(RHEL_EFIDIR_CANONICAL_PATH, os.path.basename(src_file))
+            if os.path.exists(dst_file):
+                logger.debug(
+                    "The %s file already exists in %s folder. Copying skipped."
+                    % (os.path.basename(src_file), RHEL_EFIDIR_CANONICAL_PATH)
+                )
+                continue
+
+            logger.info("Copying '%s' to '%s'" % (src_file, dst_file))
+
+            try:
+                shutil.copy2(src_file, dst_file)
+            except (OSError, IOError) as err:
+                # IOError for py2 and OSError for py3
+                self.set_result(
+                    level="ERROR",
+                    id="GRUB_FILES_NOT_COPIED_TO_BOOT_DIRECTORY",
+                    title="GRUB files have not been copied to boot directory",
+                    description=(
+                        "I/O error(%s): '%s'. Some GRUB files have not been copied to /boot/efi/EFI/redhat."
+                        % (err.errno, err.strerror)
+                    ),
+                )
+
+
+class RemoveEfiCentos(actions.Action):
+    id = "REMOVE_EFI_CENTOS"
+    dependencies = ("COPY_GRUB_FILES",)
+
+    def run(self):
+        """Remove the /boot/efi/EFI/centos/ directory when no UEFI files remains.
+
+        The centos/ directory after the conversion contains usually just grubenv,
+        grub.cfg, .. files only. Which we copy into the redhat/ directory. If no
+        other UEFI files are present, we can remove this dir. However, if additional
+        UEFI files are present, we should keep the directory for now, until we
+        deal with it.
+        """
+        super(RemoveEfiCentos, self).run()
+
+        if systeminfo.system_info.id != "centos":
+            logger.debug("Did not perform removal of EFI files - only related to CentOS Linux.")
+            # nothing to do
+            return
+        try:
+            os.rmdir(CENTOS_EFIDIR_CANONICAL_PATH)
```

Comment:
I'm getting error when trying to remove the path. In the path are still `'grubenv', 'grub.cfg'`. When I switch `copy2` to `move` (remove the source) in `CopyGrubFiles`, the only file is `grubenv`. I think also this file can be removed, since it's mentioned in the docstring as something expected what could be removed.

Reply:
Okay we can switch it to move then, is it worth to change the action name to MoveGrubFiles?

Reply:
We have to make sure that both required files are not there any more before we commit to remove the directory. 

Do we have the confidence that the previous action will correctly move/copy the required files to the new folder and not leaving anything behind? If so, then calling `move` would be fine, I believe

Reply:
Just tested and I can see the move does take care of the grub.cfg file, we will need to also remove the leftover grubenv

Reply:
The old code base had the same issue, during configuring the bootloader `WARNING - The folder /boot/efi/EFI/centos/ is left untouched. You may remove the folder manually after you ensure there is no custom data you would need.` is printed. 

But as I found by my experiments, in the folder are just `grub.cfg` and `grubenv` which I think we expect there and if in the folder are just those two files, it's fine to remove the folder. The `grub.cfg` we can `move` and the `grubenv`... we can remove manually?

>Do we have the confidence that the previous action will correctly move/copy the required files to the new folder and not leaving anything behind?

Since the `RemoveEfiCentos` has dependency `COPY_GRUB_FILES` I believe this won't run if the copy wouldn't be successfull.

Reply:
Ah, cool! Alright, makes sense then